client/core/core_test.go

```diff
@@ -2724,10 +2832,21 @@ func TestTradeTracking(t *testing.T) {
 	tBtcWallet.redeemCoins = []dex.Bytes{redeemCoin}
 	rig.ws.queueResponse(msgjson.RedeemRoute, redeemAcker)
 	tCore.tickAsset(dc, tBTC.ID)
-	// TakerSwapCast -> MatchComplete (MakerRedeem skipped when redeem ack is received with valid sig)
-	status = <-rig.db.updateMatchChan
-	if status != order.MatchComplete {
-		t.Fatalf("wrong match status wanted %v, got %v", order.MatchComplete, status)
+	// TakerSwapCast -> MakerRedeem update (after bcasting redeem)
+	newMatchStatus, err = rig.db.waitForMatchUpdate()
+	if err != nil {
+		t.Fatalf("unexpected db error: %v", err)
+	}
+	if newMatchStatus != order.MakerRedeemed {
+		t.Fatalf("wrong match status. wanted %v, got %v", order.MakerRedeemed, newMatchStatus)
+	}
+	// MakerRedeem -> MatchComplete update (after maker's redeem ack is received with valid sig)
+	newMatchStatus, err = rig.db.waitForMatchUpdate()
```

Comment:
This is the change I alluded to in my previous comment. Broadcasting a redeem now triggers 2 db updates. The first update, sets the status to `TakerSwapCast -> MakerRedeemed` for Maker instead of the previous single db update setting the status to `TakerSwapCast -> MatchComplete`. The second db update, happens after `sendRedeemAsync` completes successfully and sets the status to `MakerRedeemed -> MatchComplete`.